src/player.cpp

```diff
@@ -9612,11 +9612,9 @@ void player::rooted()
 // Should average a point every two minutes or so; ground isn't uniformly fertile
 // Overfiling triggered hibernation checks, so capping.
 {
-    double shoe_factor = footwear_factor();
-    if( (has_trait("ROOTS2") || has_trait("ROOTS3")) &&
-        g->m.has_flag("DIGGABLE", posx(), posy()) &&
-        !shoe_factor ) {
-        if( one_in(20 / shoe_factor) ) {
+    double const shoe_factor = footwear_factor();
+    if ((has_trait("ROOTS2") || has_trait("ROOTS3")) && g->m.has_flag("DIGGABLE", posx(), posy())) {
+        if( one_in(20 * (1 + shoe_factor)) ) {
```

Comment:
Not quite what is desired, though I don't the existing code does it either.
Pretty sure what we want is:
No shoes: 1/20 change of feeding.
One shoe: 1/40 chance of feeding
Two shoes: no chance of feeding.